src/java/org/apache/cassandra/service/accord/store/StoredSet.java

```diff
@@ -179,11 +179,17 @@ public long estimatedSizeOnHeap(ToLongFunction<T> measure)
         return size;
     }
 
-    public static class Navigable<T extends Comparable<?>> extends StoredSet<T, NavigableSet<T>>
+    public interface Changes<T>
```

Comment:
Pretty weird interface to stash here when it is just used in AsyncWriter for AsyncWriter things.

Reply:
All of the `StoredSet` things are pretty tightly coupled to `AsyncWriter`, and the implementations are `StoredSet`?

I wouldn't agonise over this personally, though, as it probably won't survive Blake's refactors (or not for long)